lib/bindings/HTTPBinding.js

```diff
@@ -251,7 +251,7 @@ function sendConfigurationToDevice(apiKey, deviceId, results, callback) {
         return function(error, response, body) {
             if (error) {
                 innerCallback(error);
-            } else if (response.statusCode !== 200) {
+            } else if (response && response.statusCode !== 200) {
```

Comment:
Not sure about this - what if there is no `response`? Do you really want to call the `innerCallback()` in L257 in this case or return an `errors.DeviceEndpointError` object

Reply:
In that case innerCallback() will be call, like before the proposed change.

Reply:
NTC- If that is what you want then this is fine by me. Just checking